docs/Register-Client-Application.md

```diff
@@ -0,0 +1,16 @@
+Client applications are registrations of the clients that will be requesting tokens. There are more details on the managed service documentation about [registering applications](https://docs.microsoft.com/azure/healthcare-apis/fhir-app-registration). In this guide we are going to cover how to registering a confidential client and public client application differs in the open-source FHIR Server compared to the Azure API for FHIR. 
```

Comment:
>Client applications are registrations of the clients that will be requesting tokens. [](start = 0, length = 84)

Only way for me to understand meaning of this sentence is to read link which goes after it. 
I would prefer approach we have in `Register-Resource-Application.md`
Something like:
```
In this tutorial, you'll learn how to register a confidential or public client application in Azure Active Directory.

A client application registration is an Azure Active Directory representation of an application that can be used to authenticate on behalf of a user and request access to resource applications. A confidential client application is an application that can be trusted to hold a secret and present that secret when requesting access tokens. Examples of confidential applications are server-side applications. Public clients are applications such as mobile applications and single page JavaScript applications that can't keep secrets confidential. The procedure is similar to registering a confidential client, but since public clients can't be trusted to hold an application secret, there's no need to add one.
```
I've just snatch sentences from various Azure documentations, but I think it looks nice(r)